chore(deps): bump typescript from 5.9.3 to 6.0.3

Retry of Dependabot #130 after merging #132 (`@types/node@25`).

## Context
Dependabot PR #130 was closed on 2026-04-17 because `tsc --noEmit` failed with:
```
src/lib/db/seed.ts(201,17): error TS2339: Property 'main' does not exist on type 'ImportMeta'.
```

The failure was testing against main at the time, which had `@types/node@20`. After #132 merged (`@types/node` 20→25), a local probe on TS 6 shows the project compiles clean:

- `bun run typecheck` — ✅ zero errors
- `bun run lint` — ✅ pass
- `bun run test` — ✅ 131/131
- `bun run build` — ✅ pass

## Task
Bump `typescript` from `^5` to `^6` in `package.json`, regenerate `bun.lock`, open PR. CI gates the merge.

--- a/package.json
+++ b/package.json
@@ -74,7 +74,7 @@
     "prettier": "^3.8.3",
     "prettier-plugin-tailwindcss": "^0.7.2",
     "tailwindcss": "^4",
-    "typescript": "^5",
+    "typescript": "^6",
     "vitest": "^4.1.4"
   },
   "ignoreScripts": [

